lib/internal/modules/esm/fetch_module.js

```diff
@@ -145,8 +145,9 @@ function fetchWithRedirects(parsed) {
   }
   const handler = parsed.protocol === 'http:' ? HTTPGet : HTTPSGet;
   const result = (async () => {
+    const extension = parsed.pathname.split('.').pop();
```

Comment:
This doesn't seem like the right way to probe it?  Wouldn't looking at the `assert { type: 'json' }; ` part be better than the filename which may or may not end in .json ?

Reply:
is it possible that does not end in .json?

Reply:
Yes.

Reply:
I changed it to check the assert type, I'm trying to come up with a meaningful test